### PR TITLE
STAR-58-Modal - feature(Modal): ability to spread more props into the Modal container

### PR DESCRIPTION
**Pull request by nahumzs**
### Purpose 🚀
Partially address https://github.com/acl-services/paprika/issues/369 adding a workaround via styled-components and `as={}` prop

### Notes ✏️
This is a provisional fix, the full solution is to create a context to share state on the Modal.

### Updates 📦
- [ ] MAJOR (breaking) change to _these packages_
- [x] MINOR (backward compatible) change to _these packages_
- [ ] PATCH (bug fix) change to _these packages_

### Storybook 📕
http://storybooks.highbond-s3.com/paprika/STAR-58-Modal-render

### Screenshots 📸
_optional but highly recommended_

### References 🔗
_relevant Jira ticket / GitHub issues_

### CODE REVIEW

File: packages/Modal/stories/Modal.stories.js [hunk 1]
```diff
@@ -72,6 +72,23 @@ storiesOf("Modal", module)
...
+      onSubmit={event => {
+        alert("submit");
+        event.preventDefault();
+      }}
+    >
+      <Modal.Content>
+        <input type="text" deafultValue="your text" />
+      </Modal.Content>
+      <Modal.Footer>
+        <Button type="submit">Submit</Button>
```
Comment (alexzherdev) [inline packages/Modal/stories/Modal.stories.js]: I think `isSubmit` prop is used for this. (clicking the button currently doesn't submit the form)

Comment (nahumzs) [reply packages/Modal/stories/Modal.stories.js]: abstracted logic for the input type submits into the Button to make it work. https://codesandbox.io/s/quiet-water-8khug

File: packages/Modal/stories/Modal.stories.js [hunk 2]
```diff
@@ -72,6 +72,31 @@ storiesOf("Modal", module)
...
+        event.preventDefault();
+      }}
+    >
+      <Modal.Content>
+        <input type="text" defaultValue="your text" />
+      </Modal.Content>
+      <Modal.Footer>
+        <Button
+          isSubmit
+          onClick={() => {
```
Comment (alexzherdev) [inline packages/Modal/stories/Modal.stories.js]: This is not used I think

Comment (nahumzs) [reply packages/Modal/stories/Modal.stories.js]: if the onClick is not included the onSubmit will not work when the button gets clicked, do you prefer to have a void/noop function and remove the console? unless we want to add a new event like onClickSubmit.

Comment (alexzherdev) [reply packages/Modal/stories/Modal.stories.js]: That's surprising: we have a defaultProp for `onClick` which is `() => {}`, so it should work 🤔

Comment (nahumzs) [reply packages/Modal/stories/Modal.stories.js]: your right, my bad, is working.

File: packages/Button/stories/examples/ButtonSubmit.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  return (
+    <ButtonStory>
+      <form
+        onSubmit={event => {
+          alert("submitted");
+          event.preventDefault();
+        }}
+      >
+        <input type="text" defaultValue="press enter while focus" />
+        <Button isSubmit isSemantic={false}>
```
Comment (DanilAgafonov) [inline packages/Button/stories/examples/ButtonSubmit.js]: Excuse me, but... what is this?
image

What the reason to have `isSemantic={false}`?

As I understood, we render Button as `<span>` and then trying to emulate `<button>` behavior on `<span>`. But why?

Comment (nahumzs) [reply packages/Button/stories/examples/ButtonSubmit.js]: It is for the same reason, style conflicts that are hard to fix from some applications in our ecosystem. In this case, the submit button behaves correctly as a button when it is render as span, but was lacking the ability to fire the onSubmit event, this addresses that. 

I guess eventually someday we will be able to deprecate the isSemantic prop.

Comment (alexzherdev) [reply packages/Button/stories/examples/ButtonSubmit.js]: @DanilAgafonov 
image

**Pull request merged**